File: Client/llamadas/cuentas/tipos/proveedor.go
package tipos;

type Proveedor struct {
	Nombre      string `json:"nombre"`
	Edad        string `json:"edad"`
	Descripcion string `json:"descripcion"`
	Foto        string `json:"foto"`
	Telefono    string `json:"telefono"`
	PaginaWeb   string `json:"pagina_web"`
	Facebook    string `json:"facebook"`
	Instagram   string `json:"instagram"`
	Twitter     string `json:"twitter"`
}

func (p Proveedor) ObtenerColumnas() []string {
	return []string{
		"Nombre",
		"Edad",
		"Descripcion",
		"Foto",
		"Telefono",
		"PaginaWeb",
		"Facebook",
		"Instagram",
		"Twitter",
	}
}

func (p Proveedor) ObtenerFilas() [][]string {
	return [][]string{
		{p.Nombre, p.Edad, p.Descripcion, p.Foto, p.Telefono, p.PaginaWeb, p.Facebook, p.Instagram, p.Twitter},
	}
}

File: Tecnoverde-main/API Gateway/Auth/src/main/java/co/edu/javeriana/ms/LoadBalancer/entities/Proveedor.java
package co.edu.javeriana.ms.LoadBalancer.entities;

import jakarta.persistence.Basic;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@Entity
public class Proveedor {

    @Id
    @GeneratedValue
    private Long id;
    @Basic
    private String username;
    @Basic
    private String password;
    @Basic
    private String nombre;
    @Basic
    private String edad;
    @Basic
    private String descripcion;
    @Basic
    private String telefono;
    @Basic
    private String pagina_web;
    @ManyToOne
    private Foto foto;
    @ManyToOne
    private Redes_Sociales redesSociales;

}